jenkins/PerfCI_Chaos/03_PerfCI_Choas_VMs_Deployment/Jenkinsfile

```diff
@@ -0,0 +1,191 @@
+        GOOGLE_TOKEN_FILE = 'token.json'
+        GOOGLE_DESTINATION_PATH = '/tmp'
+        GOOGLE_DRIVE_SHARED_DRIVE_ID = credentials('perfci_google_drive_shared_drive_id')
+        RUN_ARTIFACTS_URL = credentials('perfci_run_artifacts_url')
+        QUAY_BENCHMARK_RUNNER_REPOSITORY = credentials('perfci_quay_repository')
+        KUBEADMIN_PASSWORD = readFile('/home/jenkins/.kube/kubeadmin-password').trim()
+        KUBEADMIN_PASSWORD_PATH = '/home/jenkins/.kube/kubeadmin-password'
+        KUBECONFIG_PATH = '/root/.kube/config'
+        PRIVATE_KEY_PATH = '/home/jenkins/.ssh/provision_private_key'
+        CONFIG_PATH = '/home/jenkins/.ssh/config'
+        WORKSPACE='/home/jenkins'
+        SAVE_ARTIFACTS_LOCAL = 'False'
+        ENABLE_PROMETHEUS_SNAPSHOT = 'True'
+        DELETE_ALL = 'False' // Not delete the running VMs
+        RUN_STRATEGY = 'True' // runStrategy: Always for NHC/ SNR operators
+        RUN_TYPE = 'chaos_ci'
+        BOOTSTORM_SCALE = 80
+        THREADS_LIMIT= 20
+        TIMEOUT = 8000
+        WORKSPACE_PATH = "${env.WORKSPACE}/workspace/${env.JOB_NAME}/"
+    }
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                        // Clean the Jenkins workspace
+                        echo "Cleaning Jenkins workspace"
+                        deleteDir()
+                        sh '''
+                            # Check if the image exists
+                            if [[ "$(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)" != "" ]]; then
+
+                                # Get containers using the image
+                                containers=$(sudo podman ps -a --filter ancestor=${QUAY_BENCHMARK_RUNNER_REPOSITORY} -q)
+
+                                if [[ -n "$containers" ]] ; then
+                                    echo "Stopping and removing containers using the image..."
+                                    sudo podman stop $containers
+                                    sudo podman rm -f $containers
+                                fi
+
+                                # Now remove the image
+                                sudo podman rmi -f $(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)
+                            fi
+                        '''
+                }
+            }
+        }
+
+        stage('⚙️ SET SSH key') {
+            steps {
+                script {
+                    sh 'umask 700 && mkdir -p "$WORKSPACE"/.ssh'
+                    withCredentials([file(credentialsId: 'perfci_provision_private_key_file', variable: 'PROVISION_PRIVATE_KEY_FILE')]) {
+                        sh "sudo cp ${PROVISION_PRIVATE_KEY_FILE} "$WORKSPACE"/.ssh/provision_private_key"
+                        sh "sudo chown jenkins:jenkins "$WORKSPACE"/.ssh/provision_private_key"
+                    }
+                    sh 'sudo chmod 600 $PRIVATE_KEY_PATH'
```

Comment:
Quote

Reply:
Jenkisfile is very sensitive for changes, I prefer not to change it for now

Reply:
You're almost certainly going to get away with it, but if a space somehow gets into the filename it will break.

Reply:
fixed